File: 백준/Silver/1920. 수 찾기/수 찾기.py
N = int(input())
num1 = list(map(int, input().split()))
M = int(input())
num2 = list(map(int, input().split()))

li = set(num1) #중복 제거
ans = [ ]

for i in num2:
    if i in li:
        ans.append(1)
    else:
        ans.append(0)
        
for answer in ans:
    print(answer)

File: 백준/Silver/1920. 수 찾기/수 찾기.js
let input = require("fs").readFileSync("/dev/stdin").toString().split("\n");
let N = parseInt(input[0]);
let A = new Set(input[1].split(" ").map(Number)); 
let M = parseInt(input[2]);
let B = input[3].split(" ").map(Number);

let result = [];
for (let i = 0; i < M; i++) {
  if (A.has(B[i])) {
    result.push(1);
  } else {
    result.push(0);
  }
}

console.log(result.join("\n"));
